integration/internal/cmdtest/cmd.go

```diff
@@ -0,0 +1,145 @@
+package cmdtest
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"strings"
+	"testing"
+)
+
+type Cmd struct {
+	Path   string
+	Env    []string
+	Args   []string
+	Stdin  io.Reader
+	Stdout io.Writer
+	Stderr io.Writer
+	Silent bool
+}
+
+func (cmd Cmd) WithEnv(env ...string) Cmd {
+	newEnv := make([]string, len(cmd.Env))
+	copy(newEnv, cmd.Env)
+	newEnv = append(newEnv, env...)
+
+	cmd.Env = newEnv
```

Comment:
Could this be written as

```go
cmd.Env = append(cmd.Env, env...)
```

?

Doesn't seem to mutate the original slice: https://play.golang.org/p/IrYwyrju3JV

Reply:
Appending actually will mutate if the slice has capacity for it already. It's a subtle thing that's bitten me in the past.

Here's a demonstration: https://play.golang.org/p/42GnUNPhs8h

This blog post has more details: https://blog.allegro.tech/2017/07/golang-slices-gotcha.html

So I think code like this could potentially break:

```go
cmd := someCmdWithEnvSet
fly1 := cmd.WithEnv("FOO=1")
fly2 = cmd.WithEnv("FOO=2")
```

In this case it's possible for `fly1` to end up with `FOO=2` because the underlying slice was mutated.

Reply:
Yeah fair. I figured something like that would happen, but was surprised to see my playground example succeed, so generalized incorrectly